lib/Parser/Parse.cpp

```diff
@@ -8842,7 +8863,7 @@ ParseNodePtr Parser::ParseVariableDeclaration(
                         pnodeInit->sxFnc.hint = pNameHint;
                         pnodeInit->sxFnc.hintLength = nameHintLength;
                         pnodeInit->sxFnc.hintOffset = nameHintOffset;
-
+                        pnodeThis->sxVar.pid->GetTopRef()->isFuncAssignment = true;
```

Comment:
Potentially, yes. The current logic in the parser (including this change) is trying to detect common patterns to prevent obvious cases of boxing at runtime. We may at some point shift the whole legality check from byte code gen to the parser, but that's for a later release.